src/modules/design/design.ts

```diff
@@ -36,6 +42,7 @@ export class Design {
 
   public showQuitModal: boolean;
   public showLeaveModal: boolean;
+  public selectDiagramModal: boolean;
```

Comment:
```suggestion
  public showSelectDiagramModal: boolean = false;
```
Also false-initialization for the booleans above would be nice.
If we make a habit of that we can trust booleans to be `false` or `true`(and not undefined).